src/runtime_src/core/common/api/hw_queue.cpp

```diff
@@ -478,13 +549,20 @@ class kds_device : public hw_queue_impl
   std::cv_status
   wait(const xrt_core::command* cmd, size_t timeout_ms) override
   {
-    auto pkt = cmd->get_ert_packet();
+    volatile auto pkt = cmd->get_ert_packet();
     while (pkt->state < ERT_CMD_STATE_COMPLETED) {
       // return immediately on timeout
       if (exec_wait(timeout_ms) == std::cv_status::timeout)
         return std::cv_status::timeout;
     }
 
+    // One command is completed, give the runner a chance
+    // for entering sleep at next iteration.  This variable
+    // update doesn't have to be modified under a mutex lock
+    // since the exec_wait_runner will at most complete one
+    // more loop before noticing the change.
+    --m_commands_in_flight;
```

Comment:
What if there are more than one thread sharing the same device object and one is doing submit() and one is doing wait()? Will this create a race condition?